### PR TITLE
DeletePage + Home button

### PR DESCRIPTION
**Pull request by mbehlendorf**

### CODE REVIEW

File: src/main/public/locales/de/translations.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "page_failed_to_fetch": "Die Seite konnte nicht geladen werden",
   "page-editor_no_write_permissions": "Du hast nicht genügend Berechtigungen um diese Seite zu bearbeiten",
```
Comment (sdorra) [inline src/main/public/locales/de/translations.json]: No translation for **page-header_home**?

File: src/main/js/apiclient.js [hunk 2]
```diff
@@ -52,6 +52,17 @@ class ApiClient {
...
+        const deleteOptions = {
+            method: 'DELETE',
+            body: JSON.stringify(payload),
+        };
+        const options = Object.assign(deleteOptions, fetchOptions);
+        options.headers['Content-Type'] = 'application/json';
+        return fetch(createUrl(url), options)
+            .then(this.handleCasAuthentication);
+    }
+
```
Comment (sdorra) [inline src/main/js/apiclient.js]: It looks like an exact copy of post. Please create a generic function and call it from post and delete.

File: src/main/java/com/cloudogu/smeagol/wiki/infrastructure/ScmGitPageRepository.java [hunk 3]
```diff
@@ -55,6 +55,32 @@ public ScmGitPageRepository(GitClientProvider gitClientProvider) {
...
+
+            String pagePath = pagePath(path);
+            File file = client.file(pagePath);
+            if (!file.delete()) {
+                throw new IOException("could not delete file: " + file.getPath());
+            }
+
+            Author author = commit.getAuthor();
+
+            RevCommit revCommit = client.commit(
```
Comment (sdorra) [inline src/main/java/com/cloudogu/smeagol/wiki/infrastructure/ScmGitPageRepository.java]: RevCommit is never used.

File: src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageResourceAssembler.java [hunk 4]
```diff
@@ -64,6 +65,10 @@ private Link selfLink(Page page) {
         return baseLink(page).withSelfRel();
     }
 
+    private Link deleteLink(Page page) {
+        return baseLink(page).withRel("delete");
+    }
```
Comment (sdorra) [inline src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageResourceAssembler.java]: I could not find a test for the delete link.

File: src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageController.java [hunk 5]
```diff
@@ -91,6 +88,21 @@ private Path createPathFromRequest(HttpServletRequest request, WikiId id) {
         return Path.valueOf(pathExtractor.extract(request, base));
     }
 
+    @RequestMapping(method = RequestMethod.DELETE, value = "**")
+    public ResponseEntity<Void> delete(
+            HttpServletRequest request,
+            @PathVariable("repositoryId") String repositoryId,
+            @PathVariable("branch") String branch,
+            @RequestBody CreateOrEditRequestPayload payload
```
Comment (sdorra) [inline src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageController.java]: CreateOrEditRequestPayload for a delete?

File: src/main/js/wiki/components/PageHeader.js [hunk 6]
```diff
@@ -14,20 +15,26 @@ const styles = {
 
 type Props = {
     page: any,
+    deletePage: () => void,
+    clickHome: () => void,
```
Comment (sdorra) [inline src/main/js/wiki/components/PageHeader.js]: It think we should use names like __onHomeClick__ and __onDeleteClick__

File: src/main/public/locales/de/translations.json (hunk 1 above)
Comment (mbehlendorf) [reply src/main/public/locales/de/translations.json]: added

Comment (mbehlendorf) [reply src/main/public/locales/de/translations.json]: https://github.com/mbehlendorf/smeagol/blob/52e94945148c5656c94b76834aad3f0f1a45e066/src/main/public/locales/de/translations.json#L7

File: src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageResourceAssembler.java (hunk 4 above)
Comment (mbehlendorf) [reply src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageResourceAssembler.java]: Is that ok? a2b3419b3cf4d5e1ad08bfb46e275e2ec14f5a6d

Comment (sdorra) [reply src/main/java/com/cloudogu/smeagol/wiki/infrastructure/PageResourceAssembler.java]: :+1:

**Pull request merged**